### PR TITLE
Updated peak boundaries, finding and loading

### PR DESCRIPTION
**Pull request by thijsdejong10**
Update `pick_peaks.py`, the most significant change is in `find_peak_boundaries_look_ahead` I rewrote the code such that it hopefully reads better. I expect you will not like the branched logic in the function, however, every 'branch' is directly connected to an exit condition, and the code would not function without them. I've considered some of the other methods we talked about but to me none of these would function in a more concise/readable way while also still covering all changes. But if you have ideas to improve they are welcome.

I added an additional break in `find_peak_boundaries`, we talked a bit about this earlier, turns out I was still using the diff based function but I had changed the while loop to `while diff[cursor] >= 0:` causing me to go out of bounds when there is a plateau. `while diff[cursor] > 0:` should be fine though, however you could imagine that the region runs out while the signal is decreasing, so the function could still go out of bounds, and give a peak start that is higher than the peak end, so I think the extra exit condition is good to have.

in loadings I added a `peak_boundaries_from_file` in addition I changed `peak_indices_from_file` to search in a small space around the given index to be sure it lands on a local maximum.

### CODE REVIEW

File: ChromProcess/Loading/peak_collection/peak_collection_from_csv.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from pathlib import Path
 from ChromProcess import Classes
+import sys
```
Comment (thijsdejong10) [inline ChromProcess/Loading/peak_collection/peak_collection_from_csv.py:3]: Line 28 calls sys.exit but sys was never imported

File: ChromProcess/Loading/peak/peak_from_csv.py [hunk 2]
```diff
@@ -31,16 +31,38 @@ def peak_indices_from_file(peak_file, chromatogram):
     # chromatogram time axis.
     peaks_indices = np.empty(0, dtype="int64")
     for p_rt in peak_retention_times:
-        # Find the closest retention time value in to p_rt
-        delta_rt = np.abs(chromatogram.time - p_rt)
-        idx = delta_rt.argmin()
+        delta_rt = np.abs(chromatogram.time - float(p_rt))
```
Comment (Will-Robin) [inline ChromProcess/Loading/peak/peak_from_csv.py:34]: What is the reason for forcing a float conversion here? If the value is not a float already, there's a problem further upstream from this part of the code.

File: ChromProcess/Loading/peak/peak_from_csv.py [hunk 3]
```diff
@@ -31,16 +31,38 @@ def peak_indices_from_file(peak_file, chromatogram):
...
+        nearest_value_idx = delta_rt.argmin()
+        #search the 5 nearest value for the highest peak, in case the time input was not exact
+        idx = nearest_value_idx - 2 + chromatogram.signal[nearest_value_idx-2:nearest_value_idx+2].argmax() 
         peaks_indices = np.append(peaks_indices, idx)
-
     return peaks_indices
 
+def peak_boundaries_from_file(chromatogram, peak_file):
```
Comment (Will-Robin) [inline ChromProcess/Loading/peak/peak_from_csv.py:41]: This is not used anywhere, remove it.

File: ChromProcess/Loading/peak/peak_from_csv.py (hunk 2 above)
Comment (thijsdejong10) [reply ChromProcess/Loading/peak/peak_from_csv.py:34]: The peak retention time is read in as a string, so we need to convert. Does the problem arise from the float conversion?

**Pull request closed without merging**